### ISSUE TITLE
Rethinking the Inception Architecture for Computer Vision

### ISSUE THREAD

**Issue by omoindrot** [omoindrot/papers#5]
3rd paper on the **Inception** architecture from Google
### Objective

Understand the principles which guided the creation of the **Inception architecture** and allowed to scale the model.
Currently, it is very difficult to change the model because it is difficult to understand what is important and it, and what can be changed.
## Insights
1. the information flows from 299x299x3 dimensions in the input, to 1024 before the input. We have to decrease this representation size gently in the model, to avoid any **representational bottleneck**
2. ? I think they say that when we are deep in the network, we should use activations maps with a lot of features (say >1024) to disentangle the features and learn faster. We should also provide a variety of features to produce high dimensional sparse output. In Inception, they use it in the last 8x8 feature map.
   ![8x8 feature map](https://cloud.githubusercontent.com/assets/13662086/15987158/f8138bc8-301e-11e6-9319-c0b332782e7e.png)
3. When using 3x3 or 5x5 convolutions, it works as well when adding a 1x1 bottleneck layer before which reduces the dimensionality. The result is the same because of the **correlation** between adjacent units
   ![bottleneck](https://cloud.githubusercontent.com/assets/13662086/15987160/17f1f75e-301f-11e6-98f0-e610325698c0.png)
4. When we want to add parameters, we should increase both **depth** and **width** of the network
5. Factorize 5x5 or 7x7 convolutions in multiple 3x3 convolutions

![factorization](https://cloud.githubusercontent.com/assets/13662086/15987161/2674a59c-301f-11e6-92eb-e1dcf4c230d7.png)
1. Factorize 3x3 convolutions with 1x3 and 3x1 convolutions. This works well in medium grid sizes (between 12x12 and 20x20 feature maps), especially with 1x7 followed by 7x1. In Inception they used this in the 18x18 feature map.

![18x18 features map](https://cloud.githubusercontent.com/assets/13662086/15987164/41da48fa-301f-11e6-8595-bd955595d1ce.png)
1. When reducing the size of the feature map (with pooling), we have to be careful not to introduce a representational bottleneck. To avoid big computations, Inception proposes to use both pooling (cheap, but loss of representation) and convolution (costly, but keep representation intact).
   ![downsampling](https://cloud.githubusercontent.com/assets/13662086/15987192/800e4238-3020-11e6-908f-667a3eb3bced.png)
### Model Regularization via Label Smoothing

Idea for training the network: instead of using the cross entropy with the true distribution (1 for the true label, 0 elsewhere), use a combination of the true distribution and a uniform distribution:
![capture d ecran 2016-06-11 a 22 51 39](https://cloud.githubusercontent.com/assets/13662086/15987398/16bb993c-3027-11e6-8e50-90de914b3ac1.png)

K is the number of examples, epsilon is 0.1
## Inception v2

![capture d ecran 2016-06-11 a 22 09 07](https://cloud.githubusercontent.com/assets/13662086/15987217/298c5674-3021-11e6-82b8-5880af496034.png)

**Comment by hamid54** [comment]
hi,
In Inception v2 the input size is 229*229. Is it the same in V3?

Thank you!

**Comment by omoindrot** [comment]
Yes Inception v3 uses an input size of 299*299 (not 229).

You can refer to the [TensorFlow implementation](https://github.com/tensorflow/models/blob/master/inception/inception/slim/inception_model.py#L85) of the inception v3 model.

**Comment by hamid54** [comment]
Thank you!  229 was a typo, sorry!

so, you may correct the first insight which says the input size is 224x224x3.
